### PR TITLE
Update installation instructions

### PR DESCRIPTION
**Pull request by jgabry**
@rok-cesnovar @mitzimorris @bob-carpenter Based on feedback in #351 I've made the following edits: 

* Clearly mention in README that CmdStan has to be installed, not just the R package. A link is provided to the Getting Started vignette to avoid duplicating the installation content already in the vignette. 

* Add a link to the toolchain requirements in the installation section of the Getting Started vignette. These requirements change from time to time, so I think it's better to link to the CmdStan guide (https://mc-stan.org/docs/cmdstan-guide/cmdstan-installation.html). Otherwise we'll have to change the information in multiple places every time it gets updated. 

I think I addressed the other points in #351 in my response over there. But if I'm wrong in some of my comments over there we can add more to this PR to address the additional points.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +47,15 @@ or you can install the latest development version from GitHub:
 remotes::install_github("stan-dev/cmdstanr")
 ```
 
+#### Installing CmdStan
+
+If you don't already have CmdStan installed then, in addition to installing the
+R package, it is also necessary to install CmdStan using CmdStanR's 
+`install_cmdstan()` function. A suitable C++ toolchain is also required. 
+Instructions are provided in the [_Getting started with
+CmdStanR_](https://mc-stan.org/cmdstanr/articles/cmdstanr.html) vignette.
```
Comment (rok-cesnovar) [inline README.md:56]: Maybe mention that on Windows they can double check the toolchain is installed properly with check_cmdstan_toolchain()? I know this is a part of install_cmdstan so maybe redundant,

**Comment by jgabry** [comment]
Note to self: also take Bob's suggestion and do this: 


> Rather than duplicating instructions, can the readme just point to the getting started doc with full installation instructions? Or could the install instructions be pulled out of the getting started doc into their own page. I agree that we don't want to maintain duplicate instructions---it never ends well.

**Comment by jgabry** [comment]
> Note to self: also take Bob's suggestion and do this:
> 
> > Rather than duplicating instructions, can the readme just point to the getting started doc with full installation instructions? Or could the install instructions be pulled out of the getting started doc into their own page. I agree that we don't want to maintain duplicate instructions---it never ends well.

Actually we already have that in the getting started doc!

**Comment by jgabry** [comment]
Ok I think this is all set. @rok-cesnovar I added a mention of `check_cmdstan_toolchain()` like you suggested, so I'm going to go ahead and merge this.

**Comment by jgabry** [comment]
Actually I'll regenerate the online doc and then merge

**Comment by rok-cesnovar** [comment]
Thanks!

**Pull request merged**